### PR TITLE
added pundit & all authorizations

### PR DESCRIPTION
**Pull request by eric-reale**

### CODE REVIEW

File: app/controllers/bookings_controller.rb [hunk 1]
```diff
@@ -1,37 +1,41 @@
...
     # @booking.status = "Pending owner validation"
     if @booking.save
-      redirect_to camera_bookings_path
+      redirect_to dashboard_path
     else
       render 'cameras/show'
     end
   end
 
-  def index
```
Comment (Tinitos) [inline app/controllers/bookings_controller.rb]: Index not necessary?

File: app/controllers/bookings_controller.rb [hunk 2]
```diff
@@ -1,37 +1,41 @@
...
     @camera = @booking.camera
   end
 
   def update
-    # set_booking
     # @booking.status = "Pending owner validation"
-    # booking.save!
-    # redirect_to booking_path(@booking)
+    booking.save!
+    redirect_to booking_path(@booking)
```
Comment (Tinitos) [inline app/controllers/bookings_controller.rb]: maybe it should be camera_booking_path?

**Comment by Tinitos** [comment]
How does system know, if user is authorized? Should we need to include migration-file? (See last lecture slide)

File: app/controllers/bookings_controller.rb [hunk 3]
```diff
@@ -1,37 +1,41 @@
...
   def create
     @camera = Camera.find(params[:camera_id])
     @booking = Booking.new(booking_params)
+    authorize @booking
     @booking.camera = @camera
     @booking.user = current_user
     # @booking.status = "Pending owner validation"
     if @booking.save
-      redirect_to camera_bookings_path
+      redirect_to dashboard_path
```
[the remark this answers is not in the document]
Comment (eric-reale) [reply app/controllers/bookings_controller.rb]: Dashboard for mybookings page

File: app/controllers/bookings_controller.rb (hunk 1 above)
Comment (eric-reale) [reply app/controllers/bookings_controller.rb]: Index was moved up to the top

File: app/controllers/bookings_controller.rb (hunk 2 above)
Comment (eric-reale) [reply app/controllers/bookings_controller.rb]: Yes you're right. Now that we have a dashboard we'll redirect there instead.

**Pull request merged**